tc/autotuner/options_cache-inl.h

```diff
@@ -191,11 +192,39 @@ void OptionsCache<Backend>::recordRuntime(
     }
   }
   // key does not exist, emplace a new key, value
-  store_.emplace(
+  store.emplace(
       key,
       OptionsCacheValue<Backend>{std::vector<Duration>{duration}, options});
 }
 
+} // namespace detail
+
+template <typename Backend>
+void OptionsCache<Backend>::recordRuntime(
+    const lang::CanonicalTcString& tc,
+    const std::vector<TensorInfo>& inputs,
+    const std::vector<TensorInfo>& outputs,
+    const std::string& backendStr,
+    const typename Backend::MappingOptionsType& options,
+    Duration duration) {
+  OptionsCacheKey key{tc, inputs, outputs, backendStr};
+  detail::recordRuntimeImpl<Backend>(
+      mutex, numberCacheAttempts, store_, key, options, duration);
+}
+
+template <typename Backend>
+void OptionsCache<Backend>::recordRuntime(
+    const typename Backend::ExecutorType& exec,
+    Duration duration) {
+  OptionsCacheKey key{canonicalTc(exec.tc()),
+                      exec.inputsInfo(),
+                      exec.outputsInfo(),
+                      exec.deviceName()};
+
+  detail::recordRuntimeImpl<Backend>(
+      mutex, numberCacheAttempts, store_, key, exec.options(), duration);
```

Comment:
Why do you need a `recordRuntimeImpl`?
Why can't you directly call the other `recordRuntime`?